src/de.rs

```diff
@@ -1222,39 +1272,70 @@ impl<'de, 'document> de::Deserializer<'de> for &mut DeserializerFromEvents<'de,
             }
             parse_tag(tag)
         }
+        // HACK: switch to JSON enum semantics for JSON content
+        // Robust impl blocked on https://github.com/serde-rs/serde/pull/2420
+        let is_serde_content =
+            std::any::type_name::<V::Value>() == "serde::__private::de::content::Content";
```

Comment:
Ewww.. That's a little gross if you want my opinion...

 I don't get it, what are we trying to achieve here? Do we really need to do this. I didn't find such code in [serde_json](https://github.com/serde-rs/json). How are they doing it without this piece of code? Could we steal from them? Or are we trying to add a feature which is not in `serde_json`?

Reply:
`serde_json` does not use `visit_enum` in its `Deserialize` implementation, as seen [here](https://github.com/serde-rs/json/blob/faab2e8d2fcf781a3f77f329df836ffb3aaacfba/src/value/de.rs#L22-L137).

In `serde_yaml`, however, `visit_enum` is used when the value is a `TaggedValue`. In this case, V::Value should be `serde::__private::de::content::Content`, when certain serde attributes like `#[serde(flatten)]` are applied, which can cause deserialization issues. See the reference [here](https://github.com/serde-rs/serde/blob/4b3178b053683b8e13f9f551e2a40fbe4a927e43/serde/src/private/de.rs#L528-L536).

So I think it is a HACK or a work around for serde. They track related issue at: https://github.com/serde-rs/serde/issues/1183.
* I will prefix the comment with `// TODO:`.